File: LeetCode/April Challenge/12th Day/Day_12.cpp
class Solution {
public:
    int lastStoneWeight(vector<int>& stones) {
     while(stones.size()>1)
     {
         sort(stones.begin(),stones.end());
         int n=stones.size();
         cout<<n;
         int a=stones[n-1];
         int b=stones[n-2];
         stones.pop_back();
         stones.pop_back();
         if(a!=b)
             stones.push_back(a-b);
     }
        if(stones.size()==0)
            return 0;
        else
            return stones[0];
    }
};

File: LeetCode/April Challenge/12th Day/Day_12.py
import heapq
class Solution:
    def lastStoneWeight(self, stones: List[int]) -> int:
        while len(stones) > 1:
            x = heapq.nlargest(2, stones)
            if x[0] == x[1]:
                stones.remove(x[0])
                stones.remove(x[1])
            else:
                stones.remove(x[0])
                stones.remove(x[1])
                stones.append(x[0] - x[1])
        else:
            if stones: 
                return stones[0]
            else:
                return 0
        
